helpers/test/dom.test.js

```diff
@@ -757,4 +759,54 @@ describe('dom', () => {
 
 	});
 
+	describe('querySelectorComposed', () => {
+
+		const slottedElement = defineCE(
+			class extends LitElement {
+				render() {
+					return html`<div class="shadowDiv"><slot></slot></div>`;
+				}
+			},
+		);
+
+		const nestedElement = defineCE(
+			class extends LitElement {
+				render() {
+					return html`<d2l-button-icon icon="tier1:delete" text="delete"></d2l-button-icon>`;
```

Comment:
I didn't want to use button-icon here as there's a bit of leakage, but unfortunately you can't combine two `defineCE` components -- like one can't reference another.

Reply:
Would this not work:

- `nestedElement` rendered just a div with a class like `.nested`
- in the test, you did ``await fixture(`<${slottedElement}><${nestedElement}></${nestedElement}></${slottedElement}>`)``
- then querying for `.nested`?

Reply:
In that case, `nestedElement` would be inside `slottedElement`'s light DOM, so would be found as part of the first `node.querySelector` call. For this test, I want it to query a custom element's shadow root, not find it but find another custom element and then recurse and find it as part of that nested element's shadow root.

Reply:
Right, gotcha. That's fine then, though would it be an option to actually use `customElements.define()` to create an element that can be nested, and including it as a sibling to the slot in `slottedElement`? Not sure the specific benefits/requirements to using `defineCE`, apart from it seeming to create a nice impermanent element.

Reply:
Good call! Yep, that worked. Thanks for the suggestion. 🎉